src/test/java/io/github/spark_redshift_community/spark/redshift/InMemoryS3AFileSystem.java

```diff
@@ -192,23 +195,29 @@ private boolean isDir(Path f) throws IOException{
 
 
     @Override
-    public  S3AFileStatus getFileStatus(Path f) throws IOException {
+    public S3AFileStatus getFileStatus(Path f) throws IOException {
 
         if (!exists(f)) throw new FileNotFoundException();
+        Tristate state = Tristate.fromBool(dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f)));
 
         if (isDir(f)) {
-            return new S3AFileStatus(
-                true,
-                dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f)),
-                f
+            FileStatus fileStatus = new FileStatus(
+                dataMap.tailMap(toS3Key(f)).size(), true, 1,
+                this.getDefaultBlockSize(), System.currentTimeMillis(), f
+            );
+            return S3AFileStatus.fromFileStatus(
+                fileStatus, Tristate.fromBool(
```

Comment:
it would be nice to use kwargs here to improve readability

Reply:
I am not sure I follow. I modified the code to improve readability. Let me know if this makes sense.